src/test/java/api/requests/scenarios/LoanDueDatesAfterRecallTests.java

```diff
@@ -0,0 +1,398 @@
+    final IndividualResource requestServicePoint = servicePointsFixture.cd1();
+    final IndividualResource steve = usersFixture.steve();
+    final IndividualResource jessica = usersFixture.jessica();
+
+    final LoanPolicyBuilder canCirculateRollingPolicy = new LoanPolicyBuilder()
+        .withName("Can Circulate Rolling With Recalls")
+        .withDescription("Can circulate item With Recalls")
+        .rolling(Period.weeks(3))
+        .unlimitedRenewals()
+        .renewFromSystemDate()
+        .withRecallsMinimumGuaranteedLoanPeriod(Period.weeks(2))
+        .withRecallsRecallReturnInterval(Period.months(2));
+
+    final IndividualResource loanPolicy = loanPoliciesFixture.create(canCirculateRollingPolicy);
+
+    useLoanPolicyAsFallback(loanPolicy.getId(),
+        requestPoliciesFixture.noAllowedTypes().getId(),
+        noticePoliciesFixture.activeNotice().getId());
+
+    final IndividualResource loan = loansFixture.checkOut(smallAngryPlanet, steve, DateTime.now(DateTimeZone.UTC));
+
+    final String originalDueDate = loan.getJson().getString("dueDate");
+
+    requestsFixture.placeHoldShelfRequest(smallAngryPlanet, jessica,
+        DateTime.now(DateTimeZone.UTC), requestServicePoint.getId(), "Recall");
+
+    final JsonObject storedLoan = loansStorageClient.getById(loan.getId()).getJson();
+
+    assertThat("due date is not the original date",
+        storedLoan.getString("dueDate"), not(originalDueDate));
+
+    final String expectedDueDate = new DateTime(ZonedDateTime.now(clock).toInstant().toEpochMilli(), DateTimeZone.UTC).plusMonths(2).toString(ISODateTimeFormat.dateTime());
+    assertThat("due date is the RD (2 months)",
+        storedLoan.getString("dueDate"), is(expectedDueDate));
+  }
+
+  @Test
+  public void recallRequestWithMGDAndRDValuesChangesDueDateToMGD()
+      throws InterruptedException,
+      ExecutionException,
+      TimeoutException,
+      MalformedURLException {
+    final IndividualResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    final IndividualResource requestServicePoint = servicePointsFixture.cd1();
+    final IndividualResource steve = usersFixture.steve();
+    final IndividualResource jessica = usersFixture.jessica();
+
+    final LoanPolicyBuilder canCirculateRollingPolicy = new LoanPolicyBuilder()
+        .withName("Can Circulate Rolling With Recalls")
+        .withDescription("Can circulate item With Recalls")
+        .rolling(Period.weeks(3))
+        .unlimitedRenewals()
+        .renewFromSystemDate()
+        .withRecallsMinimumGuaranteedLoanPeriod(Period.weeks(2))
+        .withRecallsRecallReturnInterval(Period.weeks(1));
+
+    final IndividualResource loanPolicy = loanPoliciesFixture.create(canCirculateRollingPolicy);
+
+    useLoanPolicyAsFallback(loanPolicy.getId(),
+        requestPoliciesFixture.noAllowedTypes().getId(),
+        noticePoliciesFixture.activeNotice().getId());
+
+    // We use the loan date to calculate the MGD
```

Comment:
It's great to have comments but where should a person look for the description of the reduction?
I would use either the full name of MGD or a reference where the description is located. For example
`// We use the loan date to calculate the MGD (see doc for LoanDueDatesAfterRecallTests)`

Sorry I didn't see the names of method. 😄 For this case it will be better to put a javaDoc before each test either with reference to the main javadoc (above the class) or just copy/past javaDoc with full description. 
A developer who will be edit/extend this test class, chances are, he will be led this terms and conditions.

Reply:
There is JavaDoc at the class level that defines what MGD and RD mean. I think, for such a small unit test, that this should be good enough. I also update the comments and asserts to include the full names instead of the story defined acronyms.

Reply:
Agreed with adding the comment for reference.